examples/next-openai/ai/get_current_weather.ts

```diff
@@ -0,0 +1,51 @@
+"use server";
+import { ChatRequest, Message, nanoid } from "ai";
+import { z } from "zod";
+import { ChatCompletionRequestMessageFunctionCall } from "openai-edge";
+
+const inputSchema = z.object({
+  location: z.string(),
+  format: z.enum(["fahrenheit", "celsius"]),
+});
+
+export default async function getCurrentWeather(
+  chatMessages: Message[],
+  functionCall: ChatCompletionRequestMessageFunctionCall
+): Promise<ChatRequest> {
+  if (typeof window === "undefined") {
+    console.log("This code is running on the server side");
+  } else {
+    console.log("This code is running on the client side");
```

Comment:
🤔 if this has `use server` should it ever run on the client?

Reply:
This is just for demo purposes so show where the logs show up. You are right the client log should never trigger.

Reply:
Could you make that explicit?

Reply:
Done, however, I can remove it altogether if you don't think it's useful.